routes/project/region.rb

```diff
@@ -0,0 +1,111 @@
+# frozen_string_literal: true
+
+class Clover
+  hash_branch(:project_prefix, "region") do |r|
+    r.get true do
+      puts "project: #{@project.path}"
+      regions = @project.aws_location_credentials
+
+      if api?
+        Serializers::Region.serialize(regions)
+      else
+        @regions = Serializers::Region.serialize(regions)
+        view "region/index"
+      end
+    end
+
+    r.post true do
+      puts "geldieem project: #{@project.path}"
+      authorize("Region:create", @project.id)
+      required_parameters = ["name", "aws_region_name", "aws_access_key", "aws_secret_key"]
+      request_body_params = validate_request_params(required_parameters)
+      unless @project
+        if api?
+          fail ValidationError.new("Project not found")
+        else
+          r.redirect "region/create"
+        end
+      end
+
+      if @project.accounts_dataset.where(Sequel[:accounts][:id] => current_account_id).empty?
+        fail Authorization::Unauthorized
+      end
+
+      pl = DB.transaction do
+        alc = AwsLocationCredential.create_with_id(
+          access_key: request_body_params["aws_access_key"],
+          secret_key: request_body_params["aws_secret_key"],
+          region_name: request_body_params["aws_region_name"],
+          project_id: @project.id
+        )
+        Location.create_with_id(
+          display_name: request_body_params["name"].downcase.tr(" ", "-"),
+          name: "#{@project.ubid}-aws-#{request_body_params["aws_region_name"]}",
+          ui_name: request_body_params["name"],
+          visible: false,
+          provider: "aws",
+          aws_location_credential_id: alc.id
+        )
+      end
+
+      if api?
+        Serializers::Region.serialize(pl)
+      else
+        r.redirect "#{@project.path}/region"
+      end
+    end
+
+    r.get(web?, "create") do
+      authorize("Region:create", @project.id)
+      @available_aws_regions = ["us-east-1", "us-west-1"]
+      view "region/create"
+    end
+
+    r.on String do |region_ubid|
+      @region = AwsLocationCredential.from_ubid(region_ubid)
```

Comment:
```suggestion
      @region = @project.aws_location_credentials_dataset.first(id: UBID.to_uuid(region_ubid))
```

You should check the region is valid for the current project.